keepalived-vip/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+var (
+	flags        = pflag.NewFlagSet("", pflag.ContinueOnError)
+	resyncPeriod = flags.Duration("resync-period", 60*time.Second, `keepalived config file resync period`)
+)
+
+func init() {
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+	go wait.Until(glog.Flush, 10*time.Second, wait.NeverStop)
+}
+
+const (
+	keepalivedTmpl = "keepalived.tmpl"
+	keepalivedCfg  = "/etc/keepalived/keepalived.conf"
+
+	IngressVIPAnnotationKey = "ingress.alpha.k8s.io/ingress-vip"
```

Comment:
> Should be alpha now

Yeah, it's exactly alpha

>  this should be something like: ingress.alpha.caicloud.io?

Considering that we will contribue this to upstream, so I use "k8s.io". If we modify this, we need update our kubernetes-admin too.